### DISCUSSION TITLE
Invalid controller class: display J5

### DISCUSSION THREAD

**Discussion by astatonn** [joomla/joomla-cms#42779] [Q&A] [answered]
Hello everyone!

I'm trying to develop my onw component, however I got the message error

> Invalid controller class: display

I'm following the steps to create component on J4 and the structure of my folders is like the image attached
![image](https://github.com/joomla/joomla-cms/assets/57077913/ce0d2e83-b696-4dfb-a42d-83b967a98d0b)

My file aniversariantes.xml is like:
`
<?xml version="1.0" encoding="utf-8"?>
<extension type="component" method="install">

    <name>Aniversariantes</name>

    <creationDate>08/02/2024</creationDate>
    
    <author>3º Sgt Souza Lima</author>
    <authorUrl>github.com/astatonn</authorUrl>
    <copyright>Lucas de Souza Silva Lima</copyright>
    <license>GPL v3</license>
    <version>1.0.0</version>

    <description>
        Componente para gerenciar os aniversariantes da OM
    </description>

    <namespace path="src">Joomla\Component\Aniversariantes</namespace>
    
    <files folder="site/">
        <folder>src</folder>
        <folder>tmpl</folder>
    </files>

    
    <install>
        <sql>
            <file driver="mysql" charset="utf8">sql/install.mysql.utf8.sql</file>
        </sql>
    </install>

    <administration>
        
        <menu link="index.php?option=com_aniversariantes">Aniversariantes</menu>

        <files folder="admin/">
            <folder>services</folder>
            <folder>src</folder>
            <folder>tmpl</folder>
            <folder>sql</folder>
        </files>
    </administration>

</extension>
`

My admin/src/Controller/AniversariantesController.php is like:

```
<?php

namespace SouzaLima\Component\Aniversariantes\Administrator\Controller;

defined('_JEXEC') or die;

use Joomla\CMS\MVC\Controller\BaseController;

/**
 * @package     Joomla.Administrator
 * @subpackage  com_aniversariantes
 *
 * @copyright   Copyright (C) 2024 Lucas de Souza Silva Lima. All rights reserved.
 * @license     GNU General Public License version 3; see LICENSE
 */

/**
 * Default Controller of Aniversariantes component
 *
 * @package     Joomla.Administrator
 * @subpackage  com_aniversariantes
 */
class AniversariantesController extends BaseController {
    /**
     * The default view for the display method.
     *
     * @var string
     */
    protected $default_view = 'aniversariantes';
    
    public function display($cachable = false, $urlparams = array()) {
        return parent::display($cachable, $urlparams);
    }
    
}
```

my admin/services/provider.php is like:

```
<?php

defined('_JEXEC') or die;

use Joomla\CMS\Dispatcher\ComponentDispatcherFactoryInterface;
use Joomla\CMS\Extension\ComponentInterface;
use Joomla\CMS\Extension\MVCComponent;
use Joomla\CMS\Extension\Service\Provider\ComponentDispatcherFactory;
use Joomla\CMS\Extension\Service\Provider\MVCFactory;
use Joomla\CMS\MVC\Factory\MVCFactoryInterface;
use Joomla\DI\Container;
use Joomla\DI\ServiceProviderInterface;

return new class implements ServiceProviderInterface {
    
    public function register(Container $container): void {
        $container->registerServiceProvider(new MVCFactory('\\Joomla\\Component\\Aniversariantes'));
        $container->registerServiceProvider(new ComponentDispatcherFactory('\\Joomla\\Component\\Aniversariantes'));
        $container->set(
            ComponentInterface::class,
            function (Container $container) {
                $component = new MVCComponent($container->get(ComponentDispatcherFactoryInterface::class));
                $component->setMVCFactory($container->get(MVCFactoryInterface::class));

                return $component;
            }
        );
    }
};
```

I already search in internet, discussions topics, stackoverflow, chatgpt, gemini and can't solve this. Anyone can help me?? pleeeease

**Comment by bembelimen** [comment, marked as answer]
You always need a default "DisplayController" if you follow the convention over configuration approach. Check some core components to see that file in the "Controller" folder.

**Reply by astatonn** [reply]
Thank you. It works!
